src/rust/ide/src/transport/web.rs

```diff
@@ -174,51 +318,45 @@ impl WebSocket {
 
     /// Checks the current state of the connection.
     pub fn state(&self) -> State {
-        State::query_ws(&self.ws)
+        State::query_ws(&self.model.borrow().ws)
+    }
+
+    fn with_mut_model<R>(&mut self, f:impl FnOnce(&mut Model) -> R) -> R {
+        with(self.model.borrow_mut(), |mut model| f(model.deref_mut()))
     }
```

Comment:
I would name it `with_borrow_mut_model`. This is a to very safe utility, its worth mentioning in its name that it is borrowing the model